### PR TITLE
Dashing pep fixes

### PR DESCRIPTION
**Pull request by gonzodepedro**
The idea is to bring some changes I made into this branch

### CODE REVIEW

File: package.xml [hunk 1]
```diff
@@ -23,7 +23,6 @@
   <depend>ament_index_python</depend>
   <depend>python3-yaml</depend>
   <test_depend>ament_lint_auto</test_depend>
-  <test_depend>ament_cmake_pytest</test_depend>
```
Comment (rhaschke) [inline package.xml:26]: Why did you remove this dependency?

File: xacro/__init__.py [hunk 2]
```diff
@@ -48,7 +48,6 @@
 # Dictionary of substitution args
 substitution_args_context = {}
 
-
```
Comment (rhaschke) [inline xacro/__init__.py:51]: Doesn't PEP8 require these two line breaks?

File: xacro/__init__.py [hunk 3]
```diff
@@ -143,7 +142,6 @@ def __init__(self):
         self.defaultmap = {}  # default parameter values
         self.history = []  # definition history
 
-
```
Comment (rhaschke) [inline xacro/__init__.py:146]: Same here.

File: xacro/__init__.py [hunk 4]
```diff
@@ -298,7 +296,7 @@ def next(self):
 
 
 all_includes = []
-include_no_matches_msg = """Include tag's filename spec \"{}\" matched no files."""
+include_no_matches_msg = 'Include tag\'s filename spec "{}" matched no files.'
```
Comment (rhaschke) [inline xacro/__init__.py:299]: I'm not sure why do you want to change most of the quotes. I don't see a benefit in these changes if they don't simplify anything. Here, for example, you just move the backslash from the double to the single quote(s).

**Review by rhaschke — COMMENTED** [review]
Thanks for continuing your work on xacro. I extracted all acceptable changes in #222 and merged them.
For all remaining changes (mostly changing quotes), I would like to get a better motivation from you.
Notice, that by accident you introduced a breaking change (using `unicode` instead of `str`).

Please, in future, avoid pushing PR branches to the main repo directly, but create your own fork.
All branches created on the main repo, introduce additional noise on Travis CI.

File: xacro/__init__.py (hunk 4 above)
Comment (clalancette) [reply xacro/__init__.py:299]: The only reason to change most of the quotes is that one of the default ament python linters insists on single-quotes for most everything.  If you don't think that is worthwhile, then we won't be able to enable the linters.  If that's the case, that's fine, we can just drop most of this PR (though I still think it would be nice to remove the Python 2 fallback code).  Let us know what you want to do here.

Comment (rhaschke) [reply xacro/__init__.py:299]: I already merged the uncritical parts of this PR, particularly the removal of the Python 2 fallback code. Regarding quotes, I'm hesitating to change all the code and risk new issues. If the ament linter is too restrictive, we shouldn't use it. Apart from the quotes there are some other pep8 issues (too long lines, because the xacro code is formated for 127 chars line width, while the ament linter checks for 99 chars only), which I don't want to resolve.

Comment (clalancette) [reply xacro/__init__.py:299]: All right, that's totally fine with me.  In that case, I think we can close this out.
